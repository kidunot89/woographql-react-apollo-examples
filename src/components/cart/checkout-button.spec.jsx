import React from 'react';
import '@testing-library/react/dont-cleanup-after-each';
import { render, cleanup, waitFor } from '@testing-library/react';

import ApolloMockProvider from '../../testing/apollo-mock-provider';
import SessionProvider from '../session-provider';
import CheckoutButton from './checkout-button';

describe('CheckoutButton', () => {
  afterAll(() => cleanup());
  const { baseElement, getByText } = render(
    <ApolloMockProvider>
      <SessionProvider>
        <CheckoutButton />
      </SessionProvider>
    </ApolloMockProvider>,
  );

  it('should render successfully', async () => {
    await waitFor(() => expect(getByText(/Checkout/)).toBeTruthy());
    expect(baseElement).toMatchSnapshot();
  });
});
