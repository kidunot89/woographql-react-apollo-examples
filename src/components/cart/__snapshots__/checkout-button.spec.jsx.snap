// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutButton should render successfully 1`] = `
<body>
  <div>
    <a
      class="primary-button mx-6 py-3 w-full text-center text-xl mb-5"
      href="http://example.com/checkout/checkout?session_id=t_470b1dc6f84165a355ebde33510e53&_wpnonce=629602a9cb"
    >
      <i
        class="fa-solid fa-cash-register mr-5"
      />
      Checkout
    </a>
  </div>
</body>
`;
